src/Microsoft.OpenApi.Readers/OpenApiStreamReader.cs

```diff
@@ -42,6 +44,9 @@ public OpenApiDocument Read(Stream input, out OpenApiDiagnostic diagnostic)
         {
             var reader = new StreamReader(input);
             var result = new OpenApiTextReaderReader(_settings).Read(reader, out diagnostic);
+
+            diagnostic.HashCode = OpenApiDocument.GenerateHashValue(input);
+
```

Comment:
Actually why is this property in diagnostics at the first place?  Moving it to the document would make more sense IMHO

Reply:
The initial idea was to store the hash value here to keep track of it after reading in the document before serialization but I can move it to the OpenApi document object model

Reply:
I think the scenario we're trying to cover here is :
1. read the document
2. generate a hash
3. (store the hash somewhere)
4. read the document again
5. compare the hash with the previously stored one to know whether the document has changed since the last time

I don't think that involves the serialization at any point (beside leveraging it for hashing purpose). Does that make sense?